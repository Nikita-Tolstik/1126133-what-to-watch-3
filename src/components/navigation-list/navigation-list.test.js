import React from 'react';
import renderer from 'react-test-renderer';
import NavigationList from './navigation-list.jsx';
import {Provider} from 'react-redux';
import configureStore from "redux-mock-store";
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const mockFilm = {
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald2`,
  posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  description: `Fantastic Beasts: The Crimes of Grindelwald`,
  rating: 5,
  scoresCount: 7,
  director: `Wes Andreson`,
  stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
  runTime: 55,
  genre: `Action`,
  released: 44,
  isFavorite: true,
  videoLink: `https://upload.wikimedia.org/`,
  videoPreview: `https://upload.wikimedia.org/`,
};

const mockComments = [
  {
    id: 1,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  },
  {
    id: 2,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  },
  {
    id: 3,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  }
];

const TabType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

it(`Render NavigationList - active Tab = OVERVIEW`, () => {
  const tree = renderer.create(
      <NavigationList

        film={mockFilm}
        activeValue={TabType.OVERVIEW}
        onElementClick={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});

it(`Render NavigationList - active Tab = DETAILS`, () => {
  const tree = renderer.create(
      <NavigationList

        film={mockFilm}
        activeValue={TabType.DETAILS}
        onElementClick={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});

it(`Render NavigationList - active Tab = REVIEWS with comments`, () => {
  const store = mockStore({
    [NameSpace.COMMENT]: {
      comments: mockComments,
    }
  });

  const tree = renderer.create(

      <Provider store={store}>
        <NavigationList

          film={mockFilm}
          activeValue={TabType.REVIEWS}
          onElementClick={() => {}}
        />
      </Provider>
  );

  expect(tree).toMatchSnapshot();
});

it(`Render NavigationList - active Tab = REVIEWS without comments`, () => {
  const store = mockStore({
    [NameSpace.COMMENT]: {
      comments: [],
    }
  });

  const tree = renderer.create(

      <Provider store={store}>
        <NavigationList

          film={mockFilm}
          activeValue={TabType.REVIEWS}
          onElementClick={() => {}}
        />
      </Provider>
  );

  expect(tree).toMatchSnapshot();
});
