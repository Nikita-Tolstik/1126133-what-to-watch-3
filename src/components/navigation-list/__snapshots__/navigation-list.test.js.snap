// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavigationList - active Tab = DETAILS 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
      onClick={[Function]}
    >
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Wes Andreson
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Bill Murray,
          <br />
          Edward Norton,
          <br />
          Jude Lawfalse
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          55m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Action
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          44
        </span>
      </p>
    </div>
  </div>,
]
`;

exports[`Render NavigationList - active Tab = OVERVIEW 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
      onClick={[Function]}
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        7
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    Fantastic Beasts: The Crimes of Grindelwald
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Wes Andreson
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Bill Murray, Edward Norton, Jude Law
      </strong>
    </p>
  </div>,
]
`;

exports[`Render NavigationList - active Tab = REVIEWS with comments 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
      onClick={[Function]}
    >
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Unfortunately we don't have a reliable way.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Sophie
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              March 8, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Unfortunately we don't have a reliable way.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Sophie
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              March 8, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Unfortunately we don't have a reliable way.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Sophie
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              March 8, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Render NavigationList - active Tab = REVIEWS without comments 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
      onClick={[Function]}
    >
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item false"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <p
    style={
      Object {
        "color": "#252525",
      }
    }
  >
    Sorry. At this moment no comments yet.
  </p>,
]
`;
